projects/architecture/interfaces/primitives/blocks.proto

```diff
@@ -1,21 +1,27 @@
 syntax = "proto3";
 
 import "transactions.proto";
+import "addresses.proto";
 
 message BlockHeader {
   uint32 version = 1;
-  uint32 id = 2;
-  uint32 prevBlockId = 3;
+  bytes prevBlockHash = 2;
+  uint32 height = 3;
+}
+
+message BlockBody {
+  repeated Transaction transactions = 2;
+  repeated ModifiedStateKey stateDiff = 3;
 }
 
 message ModifiedStateKey {
-  string contractAddress = 1;
+  ContractAddress contractAddress = 1;
   string key = 2;
   string value = 3;
 }
 
 message Block {
   BlockHeader header = 1;
-  repeated Transaction transactions = 2;
-  repeated ModifiedStateKey stateDiff = 3;
+  BlockBody body = 2;
+  bytes hash = 3; // TODO: how is the hash calculated?
```

Comment:
Do we need a hash here at all? It can be quickly derived from everything (e.g., there is no block hash in either Bitcoin or Ethereum).